### ISSUE TITLE
libxml2-2.9.9 breaks a test of url_relative()

### ISSUE THREAD

**Issue by BillDunlap** [r-lib/xml2#316]
Using version 2.9.9 of libxml2.so changed uri.c:xmlBuildRelativePath() so that a test in the xml2 package (version 1.3.2) breaks:
```
url_relative("http://hadley.nz/a/c", "http://hadley.nz/") not equal to "../a/c".
1/1 mismatches
x[1]: "a/c"
y[1]: "../a/c"
```

I built that last few releases of libxml2 on a Linux box and got the following results.
```
erratic:bill:315% env LD_LIBRARY_PATH=`pwd`/libxml2-2.9.7/.libs:$LD_LIBRARY_PATH R CMD Rscript -e 'xml2::url_relative("http://foo.com/a/b/c", "http://foo.com/")'
[1] "../a/b/c"
erratic:bill:316% env LD_LIBRARY_PATH=`pwd`/libxml2-2.9.8/.libs:$LD_LIBRARY_PATH R CMD Rscript -e 'xml2::url_relative("http://foo.com/a/b/c", "http://foo.com/")'
[1] "../a/b/c"
erratic:bill:317% env LD_LIBRARY_PATH=`pwd`/libxml2-2.9.9/.libs:$LD_LIBRARY_PATH R CMD Rscript -e 'xml2::url_relative("http://foo.com/a/b/c", "http://foo.com/")'
[1] "a/b/c"
erratic:bill:318% env LD_LIBRARY_PATH=`pwd`/libxml2-2.9.10/.libs:$LD_LIBRARY_PATH R CMD Rscript -e 'xml2::url_relative("http://foo.com/a/b/c", "http://foo.com/")'
[1] "a/b/c"
```

I think both results are acceptable, although the one without the initial '..' seems better.  In any case the test should be omitted or allow both results.

**Comment by jimhester** [comment]
Yeah the particular test has a conditional on the XML version https://github.com/r-lib/xml2/blob/9b5426e5cc9ae1d4eed0ec72f92f38c2cd350916/tests/testthat/test-url.R#L21

Are you seeing the conditional not working properly for some reason?

**Comment by BillDunlap** [comment]
I think libxml2_version() gives the compile-time version, not the run-time version.

% env LD_LIBRARY_PATH=`pwd`/libxml2-2.9.9/.libs:$LD_LIBRARY_PATH R-4.0.2 CMD Rscript -e 'testthat::test_package("xml2")'
This is libxml2 version 2.9.1
── 1. Failure: url_relative (@test-url.R#22)  ──────────────────────────────────
url_relative("http://hadley.nz/a/c", "http://hadley.nz/") not equal to "../a/c".
1/1 mismatches
x[1]: "a/c"
y[1]: "../a/c"

══ testthat results  ═══════════════════════════════════════════════════════════
[ OK: 396 | SKIPPED: 4 | WARNINGS: 0 | FAILED: 1 ]
1. Failure: url_relative (@test-url.R#22)

Error: testthat unit tests failed
Execution halted

**Comment by jimhester** [comment]
Yeah, that is true
